cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -1167,6 +1254,7 @@ public RunRecordDetail recordProgramOrphaned(ProgramRunId programRunId, long end
         .setCluster(cluster)
         .setSourceId(sourceId)
         .build();
+    addFlowControlStatusKey(key, meta);
     writeToStructuredTableWithPrimaryKeys(
         key, meta, getRunRecordsTable(), StoreDefinition.AppMetadataStore.RUN_RECORD_DATA);
```

Comment:
nit: create a helper method for this two calls. (or 3 with LOG.trace). This would ensure adding status keys won't be missed and reduce duplicated code

Reply:
Added 2 liner helper function: `writeRunRecordWithPrimaryKeys()`.

Removed: `addFlowControlStatusKey()` and merged it into `writeRunRecordWithPrimaryKeys()`.

Not adding the LOG.trace as at some places the program run status is logged while at some places cluster status if logged.